syndesis-builder-image-generator/src/main/java/io/syndesis/image/Application.java

```diff
@@ -0,0 +1,184 @@
+                final Connector connector = (Connector) model.getData();
+                connectors.put(connector.getId().get(), connector);
+                for (final Action action : connector.getActions()) {
+                    steps.add(
+                        new SimpleStep.Builder()
+                            .stepKind("endpoint").
+                            connection(new Connection.Builder()
+                                .configuredProperties(map())
+                                .connectorId(connector.getId())
+                                .build())
+                            .configuredProperties(map())
+                            .action(action)
+                            .build()
+                    );
+                }
+            }
+        }
+
+        GenerateProjectRequest request = new GenerateProjectRequest.Builder()
+            .integration(new Integration.Builder()
+                .id("integration")
+                .name("Integration")
+                .description("This integration is used to prime the .m2 repo")
+                .steps( steps )
+                .build())
+            .connectors(connectors)
+            .build();
+        generate(request, project);
+    }
+
+    static private void generate(GenerateProjectRequest request, File targetDir) throws IOException {
+
+        ProjectGeneratorProperties generatorProperties = new ProjectGeneratorProperties();
+        final Map<String, String> repositories = new HashMap<>();
+        repositories.put("maven.central", "https://repo1.maven.org/maven2");
+        repositories.put("redhat.ga", "https://maven.repository.redhat.com/ga");
+        repositories.put("jboss.ea", "https://repository.jboss.org/nexus/content/groups/ea");
+
+        ConnectorCatalogProperties catalogProperties = new ConnectorCatalogProperties();
+        catalogProperties.setMavenRepos(repositories);
+
+        // We don't need to generate a real syndesis.yml, so use this dumb step registry.
+        StepVisitorFactoryRegistry registry = new StepVisitorFactoryRegistry(
+            Arrays.asList(
+                new StepVisitorFactory(){
+                    @Override
+                    public String getStepKind() {
+                        return "endpoint";
+                    }
+
+                    @Override
+                    public StepVisitor create(GeneratorContext generatorContext) {
+                        return new StepVisitor(){
+
+                            @Override
+                            public io.syndesis.integration.model.steps.Step visit(StepVisitorContext stepContext) {
+                                return null;
+                            }
+                        };
+                    }
+                }
+            )
+        );
+
+        try (InputStream is = new DefaultProjectGenerator(new ConnectorCatalog(catalogProperties), generatorProperties, registry).generate(request)) {
```

Comment:
Wouldn't it be easier to just create the pom.xml with `ProjectGenerator.generatePom()` and only use this pom.xml to download all deps ?

Reply:
you wouldn't need to deal with the tar archive then, but can directly write out this single file. We could even create simple pom.xml template on our own and write out the action gav's as deps directly so we don't have to create a dummy integration, which feels a bit hacky.

Reply:
See https://github.com/rhuss/ipaas-rest/blob/41b954a25817498ab68cb4351013ffc7d0b56bfe/project-generator/src/main/java/io/syndesis/project/converter/DefaultProjectGenerator.java#L217-L227 how the actions are extracted and https://github.com/rhuss/ipaas-rest/blob/fc03a6e4dbfd5d73760ffb4c40d30e52e272d1d3/project-generator/src/main/resources/io/syndesis/project/converter/templates/pom.xml.mustache#L74-L80 how its interpolated as dependencies.

Reply:
I could do, but I'm not sure the project would properly build via 'mvn package'

Reply:
Wouldn't a `mvn dependency:go-offline` be better anyway ?